Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -1085,6 +1081,54 @@ private extension PurchasesOrchestrator {
         }
     }
 
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    private func syncPurchasesSK2(isRestore: Bool,
+                                  initiationSource: ProductRequestData.InitiationSource,
+                                  completion: (@Sendable (Result<CustomerInfo, PurchasesError>) -> Void)?) {
+        // Don't log anything unless the flag was explicitly set.
+        let allowSharingAppStoreAccountSet = self._allowSharingAppStoreAccount.value != nil
+        if allowSharingAppStoreAccountSet, !self.allowSharingAppStoreAccount {
+            Logger.warn(Strings.purchase.restorepurchases_called_with_allow_sharing_appstore_account_false)
+        }
+
+        let currentAppUserID = self.appUserID
+        let unsyncedAttributes = self.unsyncedAttributes
+
+        self.attribution.unsyncedAdServicesToken { adServicesToken in
+            _ = Task<Void, Never> {
+                let transaction = await self.transactionFetcher.fetchLastVerifiedAutoRenewableTransaction()
+                guard let transaction = transaction, let jwsRepresentation = transaction.jsonRepresentation  else {
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completion?(.failure(ErrorUtils.transactionNotFoundError()))
```

Comment:
Should we fail if `syncPurchases` is called when user has no transactions?

Reply:
The scenario here is: if a user has no previous transactions, we cannot fetch any token, and hence we cannot post it to `/receipt` because it would return a 400 error. Instead, I fetch customer info and return it.